lib/isEmail.js

```diff
@@ -34,6 +34,7 @@ var default_email_options = {
 /* eslint-disable no-control-regex */
 var displayName = /^[a-z\d!#\$%&'\*\+\-\/=\?\^_`{\|}~\.\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+[a-z\d!#\$%&'\*\+\-\/=\?\^_`{\|}~\,\.\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF\s]*<(.+)>$/i;
 var emailUserPart = /^[a-z\d!#\$%&'\*\+\-\/=\?\^_`{\|}~]+$/i;
+var gmailUserPart = /^[a-z\d](\.?[a-z\d])+$/;
```

Comment:
This is probably vulnerable to some sort of [ReDoS](https://en.wikipedia.org/wiki/ReDoS). To sidestep this, we split the user part by `.` and then [test each part individually](https://github.com/chriso/validator.js/blob/69bdc8bbab2a7303dabce3650ec1ad044a195fe0/src/lib/isEmail.js#L73-L78). Could you set [the pattern](https://github.com/chriso/validator.js/blob/69bdc8bbab2a7303dabce3650ec1ad044a195fe0/src/lib/isEmail.js#L70-L71) we use to `/^[a-z\d]+$/i` when it's a gmail domain?

Reply:
Fixed, i did put length checking before and not inside the regex to try to avoid ReDoS but testing each part individually is a better idea since i have found that dots don't count in length checking